### PR TITLE
refactor: require password to view secret key

### LINKED ISSUE #1246: View Secret Key should require the password
Currently when "view secret key" is clicked it shows the key phrase right away (Chrome and Firefox on Windows).
image

That means if anyone has access to the computer when the wallet is not locked they can view the private key. It would be a lot more secure if this required the wallet password. 

**Environment

Version: 2.9.0
Branch: main**

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/2665249311).

Finishing up work started with: https://github.com/hirosystems/stacks-wallet-web/pull/2519

cc/ @kyranjamie @fbwoolf

### CODE REVIEW

**Comment by fbwoolf** [comment]
@kyranjamie this one has been sitting for awhile waiting to release ledger, can you review it at some point so we can merge it? I see there is a failing test, so I'll look at that today.

File: src/app/components/request-password.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { SettingsSelectors } from '@tests/integration/settings.selectors';
+import { buildEnterKeyEvent } from './link';
+import { ErrorLabel } from './error-label';
+import { PrimaryButton } from './primary-button';
+
+const waitingMessages: WaitingMessages = {
+  '2': 'Verifying password…',
+  '10': 'Still working…',
+  '20': 'Almost there.',
```
Comment (kyranjamie) [inline src/app/components/request-password.tsx]: remove full stop

File: src/app/components/request-password.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+};
+
+interface RequestPasswordProps {
+  onSuccess: (password: string) => void;
```
Comment (kyranjamie) [inline src/app/components/request-password.tsx]:
```suggestion
  onSuccess(password: string): void;
```

favour shorthand fn type notation

**Review by kyranjamie — APPROVED** [review]

File: src/app/components/request-password.tsx (hunk 2 above)
Comment (fbwoolf) [reply src/app/components/request-password.tsx]: Agree.

File: src/app/components/request-password.tsx (hunk 1 above)
Comment (fbwoolf) [reply src/app/components/request-password.tsx]: Are you saying to remove the waiting message?

Comment (kyranjamie) [reply src/app/components/request-password.tsx]: No, I'm saying the `.` is not needed for this micro copy

Comment (fbwoolf) [reply src/app/components/request-password.tsx]: Got it, haha, thx.

**Pull request merged**